addons/website_sale_delivery_mondialrelay/controllers/controllers.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+from odoo import http, _
+from odoo.addons.website_sale.controllers.main import WebsiteSale, PaymentPortal
+from odoo.exceptions import AccessDenied, ValidationError, UserError
+from odoo.http import request
+
+class Mondialrelay(http.Controller):
+
+    @http.route(['/website_sale_delivery_mondialrelay/change_billing'], type='json', auth="public", website=True, multilang=False)
+    def mondial_relay_choose(self, **data):
+        Partner_sudo = request.env['res.partner'].sudo()
+        order = request.website.sale_get_order()
+
+        if order.partner_id.id == request.website.user_id.sudo().partner_id.id:
+            raise AccessDenied('Customer of the order cannot be the public user at this step.')
+
+        p = Partner_sudo.search([
+            ('id', 'child_of', order.partner_id.commercial_partner_id.ids),
+            ('ref', '=', 'MR#%s' % data['ID']),
+            # fast check that address always the same
+            ('street', '=', data['Adresse1']),
+            ('zip', '=', data['CP']),
+        ])
+        if not p:
+            assert data['Pays'] in 'ES FR NL BE LU'.split(' '), "%s is unsupported for Mondial Relay" % data['Pays']
```

Comment:
Maybe add `MONDIAL_RELAY_COUNTRY = ['ES', 'FR', 'NL', 'BE', 'LU']` in the top of file. To inherit in the futur if new country is added.